frontend/src/components/AddTrip/index.js

```diff
@@ -1,41 +1,112 @@
 import React from 'react';
 import Button from 'react-bootstrap/Button';
+import Modal from 'react-bootstrap/Modal'
+import Form from 'react-bootstrap/Form'
 
-import '../../styles/add-trip.css'
+import createTrip from './create-new-trip.js'
 
+function createTextFormControl(placeholder, ref) {
+  return (
+    <Form.Control
+      type="text"
+      placeholder={placeholder}
+      ref={ref}
+    />
+  );
+}
+
+function createFormGroup(controlId, formLabel, inputType, placeholder, ref) {
+  let formControl;
+  if (inputType === 'text') {
+    formControl = createTextFormControl(placeholder, ref)
+  } else {
+    // TODO(Issue #..): Create diff form control for destination, time,
```

Comment:
#..?

Reply:
I was not quite ready to send this PR out to you and Keiffer yet so some things may be missing. I will create that Issue now.